pkgs/applications/science/electronics/kicad/default.nix

```diff
@@ -27,13 +27,19 @@ let
 
   wxGTK = if (stable)
     # wxGTK3x may default to withGtk2 = false, see #73145
-    then wxGTK30.override { withGtk2 = false; }
+    then wxGTK30.override {
+      withGtk2 = false;
+      withWebKit = true;
+    }
     # wxGTK31 currently introduces an issue with opening the python interpreter in pcbnew
     # but brings high DPI support?
-    else wxGTK31.override { withGtk2 = false; };
+    else wxGTK31.override {
+      withGtk2 = false;
+      withWebKit = true;
+    };
 
   python = python3;
-  wxPython = python.pkgs.wxPython_4_0;
+  wxPython = python.pkgs.wxPython_4_0.override { wxGTK = wxGTK; };
```

Comment:
this has the inconvenience that wxPython (at least currently) requires a specific revision of wxWidgets (known here as wxGTK)
optimally, the latest version of wxGTK would be the latest wxWidgets release (which it isn't, #95460)  
and wxPython would use that (which it doesn't, see #95462 (that removes `wxGTK` from its inputs because it builds its own that isn't exposed for use in kicad?))

i'm hoping this mess clears up a bit on a subsequent wxPython release, though i still expect it to be tied to a specific release (hopefully) version of wxWidgets.

Reply:
Hm? wxPython just uses wxGTK30, same as KiCad in the `stable = true` configuration.
Or do you mean that `stable = false` implies also `scripting = false`, so that KiCad would not depend on wxPython when KiCad is built with a newer wxGTK, i.e. 3.1.x, that does not build/work with wxPython 4.0?
#95460 is about wxGTK31, not wxGTK30, which is used for KiCad on macOS (at least currently).
#95462 is about wxPython 4.1, not 4.0, as used here.

Anyway, KiCad requires a specially patched version of wxWidgets, at least for 3.0.4., so a stock wxWidget won't work.

The "linear" way of doing things would be either wxPython checks for the correct version of wxWidgets or KiCad gets the wxWidgets version from wxPython (if scripting for KiCad is enabled).

Do you agree that the wxWidgets versions in KiCad and wxPython should match?

At least on macOS everything here builds, I don't know yet if the scripting interface of KiCad is actually working.

Reply:
Ah. I just saw your link to (https://gitlab.com/kicad/code/kicad/-/issues/4431) in #95462. So that answers that.